sw/device/tests/dif/dif_csrng_smoketest.c

```diff
@@ -13,87 +13,145 @@
 
 const test_config_t kTestConfig;
 
+enum {
+  kExpectedOutputLen = 16,
+};
+
 /**
- * FIPS CAVP test vector for CTR DRBG.
+ * Compare `num_words` of `got` against `expected` words.
+ *
+ * Prints differences between `got` and `expected` before generating a test
+ * assertion if a mismatch is found.
  *
- * CAVP test parameters.
- * Cipher: AES-256
- * Derivation Function: No
- * Prediction Resistance: False
+ * @param got Input buffer.
+ * @param expected Buffer containing expected values.
+ * @param num_words Number of words to compare.
  */
-const dif_csrng_seed_material_t kEntropyInput = {
-    .seed_material =
-        {
-            0xe4bc23c5,
-            0x089a19d8,
-            0x6f4119cb,
-            0x3fa08c0a,
-            0x4991e0a1,
-            0xdef17e10,
-            0x1e4c14d9,
-            0xc323460a,
-            0x7c2fb58e,
-            0x0b086c6c,
-            0x57b55f56,
-            0xcae25bad,
-        },
-    .seed_material_len = 12,
-};
+static void check_buffer(const uint32_t *got, const uint32_t *expected,
+                         size_t num_words) {
+  if (memcmp(got, expected, num_words * sizeof(uint32_t)) == 0) {
+    return;
+  }
+
+  for (uint32_t i = 0; i < num_words; ++i) {
+    LOG_INFO("[%d] got = 0x%x; expected = 0x%x", i, got[i], expected[i]);
+  }
+  CHECK(false, "Detected buffer mismatch.");
+}
 
 /**
- * Expected CSRNG output.
+ * Checks the CSRNG internal state against `expected` values.
+ *
+ * @param csrng A CSRNG handle.
+ * @param expected Expected CSRNG internal state.
  */
-const uint32_t kExpectedOutput[] = {
-    0xb2cb8905, 0xc05e5950, 0xca318950, 0x96be29ea, 0x3d5a3b82, 0xb2694955,
-    0x54eb80fe, 0x07de43e1, 0x93b9e7c3, 0xece73b80, 0xe062b1c1, 0xf68202fb,
-    0xb1c52a04, 0x0ea24788, 0x64295282, 0x234aaada,
-};
-const uint32_t kExpectedOutputLen = 16;
+static void check_internal_state(const dif_csrng_t *csrng,
+                                 const dif_csrng_internal_state_t *expected) {
+  dif_csrng_internal_state_t got;
+  CHECK(dif_csrng_get_internal_state(csrng, kCsrngInternalStateIdSw, &got) ==
```

Comment:
Shouldn't we call `wait_for_csrng_cmd_ready(csrng)` before getting the internal state?

Reply:
Done.